cpp/src/arrow/array/validate.cc

```diff
@@ -285,7 +286,7 @@ struct ValidateArrayImpl {
       const auto& field_data = *data.child_data[i];
```

Comment:
Line 275 above (FixedSizeListType) could also use validation (in the `RecurseInto` call)